### PR TITLE
Platform test enhancement

### PR DESCRIPTION
**Pull request by keboliu**
### Description of PR

Summary:
Platform test enhancement:

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### How did you do it?
Add check platform daemon status test
Add corss check sfp presence in check sfp status test
Add sysfs check for CPU, SFP, enhance the fan speed sysfs check

#### How did you verify/test it?
run platform test on Mellanox platform

### CODE REVIEW

File: tests/common/devices.py [hunk 1]
```diff
@@ -193,3 +193,33 @@ def get_crm_resources(self):
...
 
         return result
+
+    def get_pmon_daemon_list(self):
+        """
+        @summary: get pmon daemon list from the config file (/usr/share/sonic/device/{platform}/{hwsku}/pmon_daemon_control.json)
+                  if some daemon is disabled in the config file, then remove it from the daemon list.
+        """
+        full_daemon_tup = ('xcvrd', 'ledd', 'psud', 'syseepromd')
+        daemon_ctl_ley_prefix = 'skip_'
```
Comment (wangxin) [inline tests/common/devices.py]: daemon_ctl_key_prefix ?

File: tests/common/devices.py [hunk 2]
```diff
@@ -193,3 +193,33 @@ def get_crm_resources(self):
...
+        daemon_list = []
+        daemon_config_file_path = "/".join(['/usr/share/sonic/device', self.facts["platform"], 'pmon_daemon_control.json'])
```
Comment (wangxin) [inline tests/common/devices.py]: There is a better lib for this: os.path.join()

File: tests/platform/mellanox/check_sysfs.py [hunk 3]
```diff
@@ -34,46 +34,137 @@ def check_sysfs(dut):
...
+        fan_speed_get = "/var/run/hw-management/thermal/fan{}_speed_get".format(fan_id)
         try:
             fan_speed_get_content = dut.command("cat %s" % fan_speed_get)
             fan_speed = int(fan_speed_get_content["stdout"])
-            assert fan_speed > 1000, "Bad fan speed: %s" % str(fan_speed)
+            
         except Exception as e:
-            assert False, "Get content from %s failed, exception: %s" % (fan_speed_get, repr(e))
+            assert "Get content from %s failed, exception: %s" % (fan_speed_get, repr(e))
+        assert fan_speed > fan_min_speed and fan_speed < fan_max_speed, "Bad fan speed: %s" % str(fan_speed)
```
Comment (wangxin) [inline tests/platform/mellanox/check_sysfs.py]: There is a more consise format:
```
assert fan_min_speed < fan_speed < fan_max_speed, "Bad fan speed: %s" % str(fan_speed)
```

File: tests/platform/mellanox/check_sysfs.py [hunk 4]
```diff
@@ -34,46 +34,137 @@ def check_sysfs(dut):
...
+        max_tolerence_speed = ((float(fan_set_speed)/256)*fan_max_speed)*(1 + 0.3)
+        min_tolerence_speed = ((float(fan_set_speed)/256)*fan_max_speed)*(1 - 0.3)
+        assert fan_speed > min_tolerence_speed and fan_speed < max_tolerence_speed, "Speed out of tolerence speed range (%d, %d)" % (min_tolerence_speed, max_tolerence_speed)
```
Comment (wangxin) [inline tests/platform/mellanox/check_sysfs.py]: There is a more concise format:
```
assert min_tolerence_speed < fan_speed < max_tolerence_speed, "Speed out of tolerence speed range (%d, %d)" % (min_tolerence_speed, max_tolerence_speed)
```

File: tests/common/devices.py (hunk 1 above)
Comment (keboliu) [reply tests/common/devices.py]: fixed

File: tests/common/devices.py (hunk 2 above)
Comment (keboliu) [reply tests/common/devices.py]: fixed

File: tests/platform/mellanox/check_sysfs.py (hunk 3 above)
Comment (keboliu) [reply tests/platform/mellanox/check_sysfs.py]: fixed

File: tests/platform/mellanox/check_sysfs.py (hunk 4 above)
Comment (keboliu) [reply tests/platform/mellanox/check_sysfs.py]: fixed

**Pull request closed without merging**